File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
from collections import defaultdict
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ans =0
        sum =0
        dicti = defaultdict(int)
        dicti[0] =1
        for num in nums:
            sum = sum + num
            if(sum-k) in dicti:
                ans += dicti[sum-k]
            dicti[sum] +=1
        return ans

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
         mp.insert({0,1});
        int s=0,count=0;
        for(auto &a:nums){
             s += a;
            if( mp.find(s-k) != mp.end()){
                count += mp[s-k];
            }
            mp[s]++;

        }
        return count;
        
    }
};